File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int pre = 1, suf = 1, n = nums.length;
        int MAX = Integer.MIN_VALUE;
        
        for(int i=0; i<n; i++) {
            if(pre == 0) pre=1;
            if(suf == 0) suf=1;
            
            pre *= nums[i];
            suf *= nums[n-i-1];
            
            MAX = Math.max(MAX, Math.max(pre,suf));
        }
        return MAX;
    }
}

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int pre = 1, suf = 1;
        int MAX = INT_MIN;
        int n = nums.size();
        
        for(int i=0; i<nums.size(); i++) {
            if(pre == 0) pre = 1;
            if(suf == 0) suf = 1;
            
            pre *= nums[i];
            suf *= nums[n-i-1];
            MAX = max(MAX, max(pre, suf));
            
        }
        return MAX;
    }
};